src/mlpack/methods/gmm/diagonal_gmm.cpp

```diff
@@ -0,0 +1,185 @@
+/**
+ * @file diagonal_gmm.cpp
+ * @author Kim SangYeon
+ *
+ * Implementation of template-based GMM methods.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include "diagonal_gmm.hpp"
+#include <mlpack/core/math/log_add.hpp>
+
+namespace mlpack {
+namespace gmm {
+
+/**
+ * Create a DiagonalGMM with the given number of Gaussians, each of which have
+ * the specified dimensionality.  The means and covariances will be set to 0.
```

Comment:
Extra stray space here after the end of sentence.

Reply:
I'd prefer it so, but I'm not sure if other reviewers would.

Reply:
> but I'm not sure if other reviewers would.

@rcurtin, @zoq Can I get some comments about this? If everyone is fine, I'm good as well.

Reply:
Sorry for the slow response.  Typically I use two spaces; I think this can make it a little easier to read.  However that's just my personal preference and I don't feel strongly enough about that to codify it.  So, I guess, all that to say, I don't have a strong opinion here. :)

Reply:
Same, if we are all fine with it, I guess we should leave it, since it's used all over the place.

Reply:
Okay, so this is resolved now.